numpy/core/src/multiarray/descriptor.c

```diff
@@ -1711,36 +1687,32 @@ _convert_from_bytes(PyObject *obj, PyArray_Descr **at)
                 }
```

Comment:
The line just one above seems also wrong. Maybe an error in our cleanup, maybe earlier, not sure, but it should be a `return NULL;` here. I assume we do not want to replace deprecation warnings when they are raced (normally). If replacing it trips tests, maybe keep it though.

Reply:
Agreed, this looked fishy - but I wanted this patch to not change any semantics.